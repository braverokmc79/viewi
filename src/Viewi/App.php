<?php

namespace Viewi;

use Viewi\Routing\Router;
use Viewi\WebComponents\Response;

class App
{
    private static PageEngine $engine;

    public static array $config;

    public static ?array $publicConfig = null;

    public static function init(array $config, ?array $publicConfig = null)
    {
        $config[PageEngine::PUBLIC_BUILD_DIR] ??= '/viewi-build';
        self::$config = $config;
        self::$publicConfig = $publicConfig;
        self::$engine = new PageEngine(self::$config, self::$publicConfig);
    }

    public static function run(string $component, array $params)
    {
        return self::$engine->render($component, $params);
    }

    public static function getEngine(): PageEngine
    {
        return self::$engine;
    }

    public static function handle(?string $url = null, string $method = null)
    {
        $url ??= isset($_SERVER['REDIRECT_URL']) ? $_SERVER['REDIRECT_URL'] : preg_replace('/\?.*/', '', $_SERVER['REQUEST_URI']);
        $method ??= $_SERVER['REQUEST_METHOD'];
        $response = Router::handle($url, $method, $_REQUEST);
        if (is_string($response)) { // html
            header("Content-type: text/html; charset=utf-8");
            echo $response;
        } else if ($response instanceof Response) {
            http_response_code($response->StatusCode);
            foreach ($response->Headers as $name => $value) {
                header("$name: $value");
            }
            if ($response->Stringify) {
                echo json_encode($response->Content);
            } else {
                echo $response->Content;
            }
        } else { // json
            header("Content-type: application/json; charset=utf-8");
            echo json_encode($response);
        }
    }
}
