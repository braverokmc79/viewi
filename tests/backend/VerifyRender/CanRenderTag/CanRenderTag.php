<?php

use Viewi\BaseComponent;

class CanRenderTagComponent extends BaseComponent
{
    public string $title = 'Welcome to my awesome application';

    function getFullName(): string
    {
        return 'Jhon Doe';
    }

    function __init()
    {
    }
}
