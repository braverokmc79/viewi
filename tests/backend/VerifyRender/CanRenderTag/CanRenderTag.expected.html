<div>Welcome to my awesome application</div>
<p>Jhon Doe</p>